ClinGen/clincoded
# #1486: Notes for PubMed articles

**Steps to Test #1486** 

1. Begin a gene curation in the GCI
2. Add a new PMID (unless one already exists)
3. In the center PubMed box, expect to see a box below the article author information and above the abstract.
4. Expect to see 2 checkboxes and 2 text boxes (one for **Non-scorable evidence** and one for **Other comments on PMID**
5. Enter some text in either/both boxes and tick either/both boxes and click **Save.**
6. Expect feedback text to confirm that the save was successful, the save button's text to become "Edit" and for the checkboxes and text boxes to disappear. (If you didn't save any text in a text box, expect to see "None" in grey and italicized)
7. If you ticked **Non-scorable evidence**, click **Preview Evidence Summary** and expect to see the last panel to display the PMID and notes you added. (If no note, it will say "None" for the **Explanation**)
8. Return to the dashboard and expect to see in your **Recent History** "Annotation in {gene} modified for {PMID link}; {date}".
9. Click the **PMID** in the history record and expect to be navigated back to the gene curation page for that gene and expect for the previously saved notes to still be there.
10. Click **Edit** in the PubMed notes box, and expect the checkboxes and text boxes to appear.
11. Edit some of the fields and click **Save**
12. Try with multiple PMID's added. Each PubMed article should have its own set of notes.

---
**The Code**
1. The change handlers for the text boxes and checkboxes don't use refs, so I needed to modify the `Input` component to pass down `value` as a prop to `textarea`. Please let me know if it is preferable to do it another way.

2. History uses the entire `annotation` object instead of just the articleNotes. I am open to discussing other implementations.
